packages/prover/src/utils/conversion.ts

```diff
@@ -23,7 +23,8 @@ export function bufferToHex(buffer: Buffer | Uint8Array): string {
 }
 
 export function hexToBuffer(val: string): Buffer {
-  return Buffer.from(val.replace("0x", ""), "hex");
+  const hexWithEvenLength = val.length % 2 ? `0${val}` : val;
+  return Buffer.from(hexWithEvenLength.replace("0x", ""), "hex");
```

Comment:
This assumes that if the value is even length then prefix with zero.

What if the value is `0x12` and we prepend zero and make it `00x12`, which then converted to `012` and when used with Buffer the least significant byte will be dropped. 

I believe more contextual conditions to be added here and as it raises that this code might contain logical issues adding more unit tests will be nice.

Reply:
If the value is `0x12` then because `val.length % 2` is falsy, we actually don't prepend zero